### PR TITLE
(#571) Clean-up errors on startup and during sync (non-conflict)

### LINKED ISSUE #571: PushItemsAsync bug ?
### Discussed in https://github.com/Azure/azure-mobile-apps/discussions/570

<div type='discussions-op-text'>

<sup>Originally posted by **henda79** January 16, 2023</sup>
I have noticed an issue when trying to push a single table, but I'm not sure if its a bug or intended,

At the moment I have a single record in the _errors table for table A, now if I try to push table B I get an error which seems odd because the pushing of table B worked fine. It returns the error due to the previous error of table A now stored in the __errors table.

It is this code here in the PushItemsAsync method. It just gets all records from the __errors table and reports the problem.

Surely it should just get errors from the table I am trying to push ?

```
            List<TableOperationError> errors = new();
            PushStatus batchStatus = batch.AbortReason ?? PushStatus.Complete;
            try
            {
                errors.AddRange(await batch.LoadErrorsAsync(cancellationToken).ConfigureAwait(false));
            }
            catch (Exception ex)
            {
                batch.OtherErrors.Add(new OfflineStoreException("Failed to read errors from the local store.", ex));
            }
            SendPushFinishedEvent(batchStatus != PushStatus.Complete);

            // If the push did not complete successfully, then throw a PushFailedException.
            if (batchStatus != PushStatus.Complete || batch.HasErrors(errors))
            {
                List<TableOperationError> unhandledErrors = errors.Where(error => !error.Handled).ToList();
                Exception innerException = batch.OtherErrors.Count > 0 ? new AggregateException(batch.OtherErrors) : null;
                throw new PushFailedException(new PushCompletionResult(unhandledErrors, batchStatus), innerException);
            }
```</div>

### PR DESCRIPTION
**Pull request by adrianhall**
Closes #571

### CODE REVIEW

File: sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs [hunk 1]
```diff
@@ -128,10 +129,39 @@ public async Task InitializeAsync(CancellationToken cancellationToken = default)
                 // Initialize the delta token store.
                 DeltaTokenStore = new DeltaTokenStore(OfflineStore);
 
+                // Clear the errors in the store.
+                var errorsEnumerable = new FuncAsyncPageable<TableOperationError>(nextLink => GetPageOfErrorsAsync("", nextLink, cancellationToken));
+                var errors = await errorsEnumerable.ToZumoListAsync(cancellationToken).ConfigureAwait(false);
+                //await RemoveErrorsAsync(errors, cancellationToken).ConfigureAwait(false);
```
Comment (Jan-NET) [inline sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs:135]: Is the commented line 135 correct? There is no "Clear the errors..." execution! I was wondering why an error is still stuck in __error after initialization.

Comment (adrianhall) [reply sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs:135]: Ouch (for me) - yes, you are right.  I'll get a fix in this weekend (separately)

Comment (adrianhall) [reply sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs:135]: Fixed in 5.1.1

Comment (Jan-NET) [reply sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs:135]: Thanks for this very quick fix!

**Pull request merged**